### PR TITLE
Random hunger at roundstart is now controlled by adjustable amounts.

### PR DESCRIPTION
**Pull request by Blackern5000**
## About the PR
The random hungerComponent and thirstComponent in newly spawned entities is now controlled by two adjustable numbers separate from the hunger thresholds, and can be made to be a specific number instead.

## Why / Balance
Some mobs, such as mice, already use this functionality, however it has been nonfunctional. It also seems helpful for in the future if say, someone wanted to make rat king have a specific variation for it's starting hunger, or something similar.
image
The mouse hungerComponent uses currentHunger, which was not able to be set manually with the previous version

## Technical details
hungerComponent and thirstComponent now have two new variables which act as the minimum and maximum for random hunger. These numbers default to 110 min/150 max and 310 min/449 max for hunger and thirst respectively, and can be specified in the prototype or modified before mapinit. If a prototype has a specified currentHunger equal to or above 0 (defaults to -1 to allow this), it will entirely skip the random hunger variation and just use that number instead.

## Media
image

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
Some entities may have extremely high or extremely low starting hunger with the default variation, cockroaches come to mind, but I think they're parented to mice so I'm not sure.

**Changelog**
I don't think this needs a changelog

### CODE REVIEW

File: Content.Shared/Nutrition/Components/HungerComponent.cs [hunk 1]
```diff
@@ -63,6 +63,15 @@ public sealed partial class HungerComponent : Component
         { HungerThreshold.Dead, 0.0f }
     };
 
+    /// <summary>
+    /// The minimum and maximum amount that hunger can be randomly set at when an entity is spawned
+    /// </summary>
+    [DataField("hungerMin"), ViewVariables(VVAccess.ReadWrite)]
```
Comment (EmoGarbage404) [inline Content.Shared/Nutrition/Components/HungerComponent.cs:69]:
```suggestion
    [DataField]
```
do this for all datafields

File: Content.Shared/Nutrition/EntitySystems/ThirstSystem.cs [hunk 2]
```diff
@@ -50,12 +50,12 @@ public override void Initialize()
...
-        // Do not change behavior unless starting value is explicitly defined
+        // If CurrentThirst is defined already, it uses that instead.
         if (component.CurrentThirst < 0)
         {
             component.CurrentThirst = _random.Next(
-                (int) component.ThirstThresholds[ThirstThreshold.Thirsty] + 10,
-                (int) component.ThirstThresholds[ThirstThreshold.Okay] - 1);
+                (int) component.thirstVariationMin,
+                (int) component.thirstVariationMax);
         }
```
Comment (EmoGarbage404) [inline Content.Shared/Nutrition/EntitySystems/ThirstSystem.cs:59]: this really should use the thresholds or else any modification to them is going to require also updating these values for parity.

File: Content.Shared/Nutrition/Components/ThirstComponent.cs [hunk 3]
```diff
@@ -57,6 +57,15 @@ public sealed partial class ThirstComponent : Component
         {ThirstThreshold.Dead, 0.0f},
     };
 
+    /// <summary>
+    /// The minimum and maximum amount that thirst can be randomly set at when an entity is spawned
+    /// </summary>
```
Comment (EmoGarbage404) [inline Content.Shared/Nutrition/Components/ThirstComponent.cs:62]: for both of these, you need to copy the comment and use "min" or "max" when applicable. summary comments are per-variable and dont make sense when written like this.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by ps3moira** [comment]
somewhat out of this scope but would this be applicable to all crew who join the server/spawn in having random chances of hunger levels?

**Pull request closed without merging**